inference-engine/src/transformations/src/transformations/common_optimizations/random_uniform_fusion.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/random_uniform_fusion.hpp"
+
+#include <memory>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RandomUniformFusion, "RandomUniformFusion", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RandomUniformMaxValFusion, "RandomUniformMaxValFusion", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RandomUniformMinValFusion, "RandomUniformMinValFusion", 0);
+
+ngraph::pass::RandomUniformMaxValFusion::RandomUniformMaxValFusion() {
+    MATCHER_SCOPE(RandomUniformMaxValFusion);
+    auto data_pattern = ngraph::pattern::any_input();
+    auto ru_min_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto ru_max_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto random_uniform_pattern =
+        ngraph::pattern::wrap_type<opset8::RandomUniform>({data_pattern, ru_min_const_pattern, ru_max_const_pattern},
+                                                          pattern::consumers_count(1));
+    auto mul_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    auto mul_pattern = ngraph::pattern::wrap_type<opset8::Multiply>({random_uniform_pattern, mul_const_pattern});
+
```

Comment:
I didn't move Add and Mul because these transformations covered only case with min=0.0 and max=1.0. So it wasn't necessary to add 0.0 and multiply by 1.0. But in general case with arbitrary min and max values we should move Add and Mul into RandomUniform inputs. I extended transformations for this case and removed min=0.0 and max=1.0 restrictions.

Regarding second question, multiple Mul and And case is covered by other Fusing transformations.

Reply:
I added comment that transformation must be executed after LinOpSequenceFusion, as this transformation fuses Mul and Add and sets the order Mul -> Add.